src/suspendplugin.c

```diff
@@ -407,19 +460,16 @@ suspend_plugin_init()
         mce_display_add_state_changed_handler(mce_display,
             suspend_display_event, NULL);
 
-    nl_socket = nl_socket_alloc();
+    nl_socket_fd = nl_socket_get_fd(nl_socket);
+    netlink_channel = g_io_channel_unix_new(nl_socket_fd);
+    g_io_channel_set_close_on_unref(netlink_channel, TRUE);
```

Comment:
Looks like nl_socket "owns" nl_socket_fd - using g_io_channel_set_close_on_unref() is a bit suspicious.

Reply:
Besides, `g_io_channel_set_close_on_unref()` is redundant here because `g_io_channel_shutdown()` is explicitly called by `suspend_plugin_exit()` and that unconditionally closes the socket. This `g_io_channel_set_close_on_unref()` call should be removed.

Reply:
@monich are you sure? I'd assume something_*shutdown() would make the fd unusable for i/o to one/both directions, but still valid otherwise i.e. != close()

Reply:
Used the source... g_io_channel_shutdown() seems to indeed do  similar things as g_io_channel_close(). Well, color me surprised.